packages/react-reconciler/src/__tests__/useMemoCache-test.js

```diff
@@ -0,0 +1,367 @@
+let React;
+let ReactNoop;
+let act;
+let useState;
+let useMemoCache;
+let ErrorBoundary;
+
+describe('ReactCache', () => {
+  beforeEach(() => {
+    jest.resetModules();
+
+    React = require('react');
+    ReactNoop = require('react-noop-renderer');
+    act = require('jest-react').act;
+    useState = React.useState;
+    useMemoCache = React.unstable_useMemoCache;
+
+    class _ErrorBoundary extends React.Component {
+      constructor(props) {
+        super(props);
+        this.state = {hasError: false};
+      }
+
+      static getDerivedStateFromError(error) {
+        // Update state so the next render will show the fallback UI.
+        return {hasError: true};
+      }
+
+      componentDidCatch(error, errorInfo) {}
+
+      render() {
+        if (this.state.hasError) {
+          // You can render any custom fallback UI
+          return <h1>Something went wrong.</h1>;
+        }
+
+        return this.props.children;
+      }
+    }
+    ErrorBoundary = _ErrorBoundary;
+  });
+
+  // @gate experimental || www
```

Comment:
Nit: I usually prefer to gate on the feature flag, `enableUseMemoCacheHook`. That way you could change which channels a feature is available in without updating all the gate conditions.